converter/converters/pod/affinity.go

```diff
@@ -0,0 +1,398 @@
+package pod
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/golang/glog"
+	"github.com/koki/short/types"
+)
+
+func RevertAffinities(a []types.Affinity) (*v1.Affinity, error) {
+	node, pod, antiPod, err := splitAffinities(a)
+	if err != nil {
+		return nil, err
+	}
+
+	n, err := nodeAffinity(node)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := podAffinity(pod)
+	if err != nil {
+		return nil, err
+	}
+
+	ap, err := podAntiAffinity(antiPod)
+	if err != nil {
+		return nil, err
+	}
+
+	if n == nil && p == nil && ap == nil {
+		return nil, nil
+	}
+
+	return &v1.Affinity{
+		NodeAffinity:    n,
+		PodAffinity:     p,
+		PodAntiAffinity: ap,
+	}, nil
+}
+
+type PodAffinity struct {
+	Affinity   string
+	Topology   string
+	Namespaces []string
+}
+
+func splitAffinities(as []types.Affinity) (node []string, pod, antiPod []PodAffinity, err error) {
+	node = []string{}
+	pod = []PodAffinity{}
+	antiPod = []PodAffinity{}
+
+	for _, a := range as {
```

Comment:
for i := range affinities {
  affinity := affiities[i]

Reply:
I updated the variable names, but I think it's cleaner to ignore the index if all I care about is the array element itself. Is there a reason beyond aesthetic concerns to not use the `_, elem` form?